src/main/scala/AdvAnagrams.scala

```diff
@@ -0,0 +1,158 @@
+import java.util
+
+import scala.collection.mutable.{ArrayBuffer, ListBuffer, Map => MutableMap}
+import scala.io.Source
+import scala.io.StdIn.readLine
+
+object AdvAnagrams extends App
+{
+  var rawInput: ArrayBuffer[String] = takeInput()
+  var wordListMap: Map[String, String] = createWordListmap(sortwordList(generatewordList()), generatewordList())
+  //println(wordListMap)
+  //wordListMap.foreach(value => println(value))
+
+  //println(findMostAngarmsNum(createAnagramList(createSortedinput(rawInput), wordListMap) , rawInput))
+  var mostAnagramList: ArrayBuffer[String] = (createListOfMostAnagram
+  (
+    createAnagramList(createSortedinput(rawInput), wordListMap)
+    , rawInput
+    , findMostAngarmsNum(createAnagramList(createSortedinput(rawInput), wordListMap) , rawInput)
+  ))
+
+  //mostAnagramList.foreach(item => println(item))
+  findlongestWord(mostAnagramList, findlongestWordNum(mostAnagramList))
+
+  //val mostAnagramList =
+
+
+
+
+  //def createAllAnagramLists(): List[List[String]]=
+  //{
+
+  //}
+
+  def findlongestWordNum(arrayBuffer: ArrayBuffer[String]):  Int =
```

Comment:
arrayBuffer should be renamed to something like "wordList" or anything else making it more clear what the collection is holding